src/UI/Actions.hs

```diff
@@ -1298,14 +1298,20 @@ makeAttachmentsFromSelected s = do
   let toggled = view (_2 . fsEntryName) <$> toListOf (toggledItemsL (Proxy @'ListOfFiles)) s
       selected = toListOf (list (Proxy @'ListOfFiles) . to L.listSelectedElement . _Just . _2 . _2 . fsEntryName) s
   parts <- traverse (\x -> createAttachmentFromFile (mimeType x) (makeFullPath x)) (toggled `union` selected)
-  pure $ s & over (asCompose . cAttachments) (go parts)
+  pure $ s & over (asCompose . cAttachments) (listAppendAttachments parts)
     . over (asViews . vsFocusedView) (Brick.focusSetCurrent ComposeView)
     . set (asViews . vsViews . ix ComposeView . vFocus) ComposeListOfAttachments
   where
-    go :: [MIMEMessage] -> L.List Name MIMEMessage -> L.List Name MIMEMessage
-    go parts l = foldr upsertPart l parts
     makeFullPath path = currentLine (view (asFileBrowser . fbSearchPath . E.editContentsL) s) </> path
 
+listAppendAttachments :: [MIMEMessage] -> L.List Name MIMEMessage -> L.List Name MIMEMessage
+listAppendAttachments parts l =
+  let append newPart l' =
+        l' &
+        over L.listElementsL (`snoc` newPart) .
+        set L.listSelectedL (Just (view (L.listElementsL . to length) l'))
+   in foldr append l parts
```

Comment:
This seems to attach the new parts one by one.  Why not do a single `(<>)`?
```haskell
listAppendAttachments parts = listMoveTo (-1) . over L.listElementsL (<> parts)
```
Would that work?

Reply:
Ah of course! This is wonderful! Thank you so much.